dashboard-api/internal/platform/storage/user_repository.go

```diff
@@ -0,0 +1,35 @@
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	studiesmanager "github.com/vlab-research/vlab/dashboard-api/internal"
+)
+
+type UserRepository struct {
+	db *sql.DB
+}
+
+func NewUserRepository(db *sql.DB) *UserRepository {
+	return &UserRepository{
+		db: db,
+	}
+}
+
+func (r *UserRepository) CreateUser(ctx context.Context, userId string) (studiesmanager.User, error) {
+	_, err := r.db.Exec("INSERT INTO users (id) VALUES ($1)", userId)
+
+	if err != nil {
+		if err.Error() == "ERROR: duplicate key value violates unique constraint \"primary\" (SQLSTATE 23505)" {
```

Comment:
Isn't there a code you can check instead of the entire message? Might be more robust? Any unique constraint violation should suffice here, no?

Reply:
Just for context to anyone viewing, we discussed this elsewhere.

We need to identify the errors that happen due the user already exists and provide specific errors that will be used for the clients of the api.

The error code 23505 it's used when any unique constrained it's violated while processing the query. Although right now, the primary key of the table is the only column with a unique constraint, there may be more in the future.

So I decided to go with this approach of comparing the string. It seems flacky but:
- Dependencies are locked
- We'll have metrics and alerts for this endpoint
- Soon there will be e2e that integrates with the real db and will be run as part of our CI/CD pipeline.

If at some point this bite us, we can rethink how we approach it and may change the approach. For example perform an extra query that checks if the user exists before attempting to create it.